client/src/pages/NewGame.js

```diff
@@ -32,26 +33,36 @@ const NewGame = (props) => {
 
   const localData = localStorage.getItem("newGame");
 
-  // Calls API if no locally stored data, with otherwise use local data.
+  //Users joining the match
   useEffect(() => {
-    if (localData) {
+    console.log("User joins the match");
+    const socket = openSocket("http://localhost:3001");
+    socket.on("connect", () => {
+      socket.on("join-match", (data) => {
+        alert("New user joined Match");
+      });
+    });
+  });
```

Comment:
if you want to run only once on page load, you'll probably need an empty dependency here

```suggestion
  }, []);
```